cmd/secrets-provider/main.go

```diff
@@ -63,40 +63,28 @@ func main() {
 	expBackoff.MaxElapsedTime = 2 * time.Minute
 
 	err = backoff.Retry(func() error {
-		for {
```

Comment:
1. why did the `for` go away?
2. did anything else change here apart from removing the `for`?

Reply:
:one: because we don't need it since we aren't a sidecar and we don't want to restart the process since we are an init container / job.

:two: Yes we removed unreachable code 
```
if authnConfig.ContainerMode == "init" {
  os.Exit(0)
}

// Reset exponential backoff
expBackoff.Reset()

log.Info(messages.CSPFK007I, authn.Config.TokenRefreshTimeout)

fmt.Println()
time.Sleep(authn.Config.TokenRefreshTimeout)
```

Reply:
unreachable because the process hits, it will exit anyways because it is always ContainerMode init
```
if authnConfig.ContainerMode == "init" {
  os.Exit(0)
}
```

Reply:
well, this PR does a LOT more than just adding logs to the codebase.

if i understand correctly, after this change we don't use the `CONTAINER_MODE` parameter. will we not use it anytime in the future? won't we ever run as a sidecar? Also, we need to remove it from the docs in that case.

seems that `CSPFK007I` is never used.

Reply:
don't forget that the default value for `authnConfig.ContainerMode` is `sidecar`. i know that we are now not querying this value but what will happen if we choose to support sidecar in the future? how will we do so _and_ maintain backwards compatibility?

Reply:
I think we should still use `CONTAINER_MODE` / have customer's declare it because it allows us to determine which flow we will use init or Job. 

We need it for this check
```
if secretsConfig.StoreType == secretsConfigProvider.K8S && authnConfig.ContainerMode != "init" {
   printErrorAndExit(messages.CSPFK007E)
}
```
Here is where we will determine which flow the customer defined for us and limit them to either init (ContainerMode) or Job (SupportMultipleApps)

If we choose to move to sidecar deployments then we will add this back in but IMO it makes the code a lot less readable because we have places that are just not reachable.